File: app/src/main/java/com/denchic45/kts/ui/confirm/ConfirmDialog.java
package com.denchic45.kts.ui.confirm;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.ViewModelProvider;

import com.denchic45.kts.R;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;

import org.jetbrains.annotations.NotNull;

public class ConfirmDialog extends DialogFragment {

    private static final String TITLE = "TITLE";
    private static final String MESSAGE = "SUBTITLE";
    private ConfirmViewModel viewModel;

    @NotNull
    public static ConfirmDialog newInstance(String title, String subtitle) {
        ConfirmDialog fragment = new ConfirmDialog();
        Bundle args = new Bundle();
        args.putString(TITLE, title);
        args.putString(MESSAGE, subtitle);
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        MaterialAlertDialogBuilder dialog = new MaterialAlertDialogBuilder(requireActivity(), R.style.MaterialAlertDialog_Rounded)
                .setTitle(getArguments().getString(TITLE))
                .setMessage(getArguments().getString(MESSAGE))
                .setPositiveButton("Да", (dialogInterface, i) -> viewModel.onPositiveClick())
                .setNegativeButton("Нет", (dialog1, which) -> viewModel.onNegativeClick());
        return dialog.create();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        viewModel = new ViewModelProvider(this).get(ConfirmViewModel.class);
//        getDialog().setCanceledOnTouchOutside(false);
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onDismiss(@NonNull DialogInterface dialog) {
        super.onDismiss(dialog);
    }
}

File: app/src/main/java/com/denchic45/kts/ui/confirm/ConfirmDialog.kt
package com.denchic45.kts.ui.confirm

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.viewModels
import com.denchic45.kts.R
import com.denchic45.kts.di.viewmodel.ViewModelFactory
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import dagger.android.support.AndroidSupportInjection
import javax.inject.Inject

class ConfirmDialog : DialogFragment() {

    @Inject
    lateinit var viewModelFactory: ViewModelFactory<ConfirmViewModel>
    val viewModel: ConfirmViewModel by viewModels { viewModelFactory }
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog =
            MaterialAlertDialogBuilder(requireActivity(), R.style.MaterialAlertDialog_Rounded)
                .setTitle(requireArguments().getString(TITLE))
                .setMessage(requireArguments().getString(MESSAGE))
                .setPositiveButton("Да") { dialogInterface: DialogInterface?, i: Int -> viewModel.onPositiveClick() }
                .setNegativeButton("Нет") { dialog1: DialogInterface?, which: Int -> viewModel.onNegativeClick() }
        return dialog.create()
    }

    companion object {
        const val TITLE = "TITLE"
        const val MESSAGE = "SUBTITLE"
        fun newInstance(title: String?, subtitle: String?): ConfirmDialog {
            val fragment = ConfirmDialog()
            val args = Bundle()
            args.putString(TITLE, title)
            args.putString(MESSAGE, subtitle)
            fragment.arguments = args
            return fragment
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        AndroidSupportInjection.inject(this)
    }
}